src/braket/devices/local_simulator.py

```diff
@@ -204,9 +204,12 @@ def run_batch(  # noqa: C901
                         f"{unbounded_parameters}"
                     )
 
-        with Pool(min(max_parallel, len(tasks_and_inputs))) as pool:
-            param_list = [(task, shots, inp, *args, *kwargs) for task, inp in tasks_and_inputs]
-            results = pool.starmap(self._run_internal_wrap, param_list)
+        if isinstance(self._delegate, MultiSimulator):
```

Comment:
I would like to avoid this if statement, if possible. Maybe after we refactor to minimize duplication, there will be a reasonable way to get rid of it?

Reply:
Yes, I'll do my best to get rid of this. I agree, I think there should be a more elegant way.

Reply:
Use Python's multiple dispatch package? 😇

Reply:
You can also just use `@singledispatchmethod`